PyArrow broke the build on ARM platform [Regression]

`pip install apache-superset==0.35.1` on Raspbian (Raspberry pi 4, ARM) ends with
```
ERROR: Could not find a version that satisfies the requirement pyarrow<0.15.0,>=0.14.1 (from apache-superset) (from versions: 0.9.0, 0.10.0, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.13.0, 0.14.0, 0.15.1)
ERROR: No matching distribution found for pyarrow<0.15.0,>=0.14.1 (from apache-superset)

Version `0.34.1` installed correctly.
```

### Expected results

Apache superset installs correctly.

### Actual results

The installation ends with an error.

#### Screenshots

If applicable, add screenshots to help explain your problem.

#### How to reproduce the bug

1. Use system with ARM architecture.
2. Run `pip install apache-superset==0.35.1`

### Environment

(please complete the following information):

- superset version: `0.35.1`
- python version: `3.7.3`
- OS: Linux raspberrypi 4.19.75-v7l+ #1270 SMP Tue Sep 24 18:51:41 BST 2019 armv7l GNU/Linux

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I tried to installed the latest version of `pyarrow` (as I saw the recent not released bump to 0.15.1).

This fails with: `ERROR: Could not build wheels for pyarrow which use PEP 517 and cannot be installed directly`

@jaksmid Yes exactly. That was not expected since numpy had just been compiled, built and installed in my local env, and I ran it in python without an issue. 

```
  CMake Error at cmake_modules/FindNumPy.cmake:62 (message):
    NumPy import failure:

    Traceback (most recent call last):

      File "<string>", line 1, in <module>

    ModuleNotFoundError: No module named 'numpy'

  Call Stack (most recent call first):
    CMakeLists.txt:189 (find_package)


  -- Configuring incomplete, errors occurred!
  See also "/tmp/pip-install-7728kxma/pyarrow/build/temp.linux-aarch64-3.7/CMakeFiles/CMakeOutput.log".
  See also "/tmp/pip-install-7728kxma/pyarrow/build/temp.linux-aarch64-3.7/CMakeFiles/CMakeError.log".
  error: command 'cmake' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyarrow
  Running setup.py clean for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow which use PEP 517 and cannot be installed directly
```

--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -189,7 +189,7 @@ prison==0.1.3
     # via flask-appbuilder
 py==1.9.0
     # via retry
-pyarrow==3.0.0
+pyarrow==4.0.0
     # via apache-superset
 pycparser==2.20
     # via cffi

--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -189,7 +189,7 @@ prison==0.1.3
     # via flask-appbuilder
 py==1.9.0
     # via retry
-pyarrow==4.0.0
+pyarrow==4.0.1
     # via apache-superset
 pycparser==2.20
     # via cffi

--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ setup(
         "python-dateutil",
         "python-dotenv",
         "python-geohash",
-        "pyarrow>=4.0.0, <4.1",
+        "pyarrow>=4.0.1, <4.1",
         "pyyaml>=5.4",
         "PyJWT>=1.7.1, <2",
         "redis",

--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ setup(
         "python-dateutil",
         "python-dotenv",
         "python-geohash",
-        "pyarrow>=3.0.0, <3.1",
+        "pyarrow>=4.0.0, <4.1",
         "pyyaml>=5.4",
         "PyJWT>=1.7.1, <2",
         "redis",

